pilot/pkg/serviceregistry/kube/controller/ambientindex_test.go

```diff
@@ -773,104 +1046,79 @@ func addPolicy(t *testing.T, cfg *memory.Controller, name, ns string, selector m
 ) {
 	t.Helper()
-	service := generateService(name, "ns1", labels, annotations, ports, selector, ip)
-	sc.CreateOrUpdate(service)
+	service := generateService(name, testNS, labels, annotations, ports, selector, ip)
+	s.sc.CreateOrUpdate(service)
 }
 
-func assertWorkloads(t *testing.T, controller *FakeController, lookup string, state workloadapi.WorkloadStatus, names ...string) {
-	t.Helper()
-	want := sets.New(names...)
-	assert.EventuallyEqual(t, func() sets.String {
-		var workloads []*model.AddressInfo
-		if lookup == "" {
-			workloads = controller.ambientIndex.All()
-		} else {
-			workloads = controller.ambientIndex.Lookup(lookup)
-		}
-		have := sets.New[string]()
-		for _, wl := range workloads {
-			switch addr := wl.Address.Type.(type) {
-			case *workloadapi.Address_Workload:
-				if addr.Workload.Status == state {
-					have.Insert(addr.Workload.Name)
-				}
-			}
-		}
-		return have
-	}, want, retry.Timeout(time.Second*3))
+func (s *ambientTestServer) lookup(key string) []*model.AddressInfo {
+	if key == "" {
+		return s.controller.ambientIndex.All()
+	}
+	return s.controller.ambientIndex.Lookup(key)
 }
 
-func addPod(t *testing.T, pc clienttest.TestClient[*corev1.Pod], ip string, name, sa string, labels map[string]string, annotations map[string]string) {
-	t.Helper()
-	pod := generatePod(ip, name, "ns1", sa, "node1", labels, annotations)
+func (s *ambientTestServer) clearEvents() {
+	s.fx.Clear()
+}
 
-	p := pc.Get(name, pod.Namespace)
-	if p == nil {
-		// Apiserver doesn't allow Create to modify the pod status; in real world its a 2 part process
-		pod.Status = corev1.PodStatus{}
-		newPod := pc.Create(pod)
-		setPodReady(newPod)
-		newPod.Status.PodIP = ip
-		newPod.Status.PodIPs = []corev1.PodIP{
-			{
-				IP: ip,
-			},
-		}
-		newPod.Status.Phase = corev1.PodRunning
-		pc.UpdateStatus(newPod)
-	} else {
-		pc.Update(pod)
-	}
+// Returns the XDS resource name for the given pod.
+func (s *ambientTestServer) podXdsName(name string) string {
+	return fmt.Sprintf("%s//Pod/%s/%s",
+		s.controller.clusterID, testNS, name)
+}
+
+// Returns the XDS resource name for the given address.
+func (s *ambientTestServer) addrXdsName(addr string) string {
+	return string(s.controller.network) + "/" + addr
+}
+
+// Returns the XDS resource name for the given service.
+func (s *ambientTestServer) svcXdsName(serviceName string) string {
+	return fmt.Sprintf("%s/%s", testNS, s.hostnameForService(serviceName))
+}
+
+// Returns the hostname for the given service.
+func (s *ambientTestServer) hostnameForService(serviceName string) string {
+	return fmt.Sprintf("%s.%s.svc.company.com", serviceName, testNS)
+}
+
+// Returns the XDS resource name for the given WorkloadEntry.
+func (s *ambientTestServer) wleXdsName(wleName string) string {
+	return fmt.Sprintf("%s/networking.istio.io/WorkloadEntry/%s/%s",
+		s.controller.clusterID, testNS, wleName)
+}
+
+// Returns the XDS resource name for the given ServiceEntry IP address.
```

Comment:
```suggestion
// Returns the XDS resource name for the a Workload generated by a WorkloadEntry inlined on a ServiceEntry. The address delineates which workload we are referring to.
```

or something to that effect might be clearer. especially since the address doesn't have to be an IP; its the address on the WE which could be a hostname